File: Exercism Codes/JavaScript/rotational-cipher.js
export const rotate = (text, key) => {
  return text.split("").map(x => rotateLetter(x, key)).join("");
};

const rotateLetter = (x, key) => {
  return /[A-Za-z]/.test(x) ? String.fromCharCode((((x.charCodeAt(0) - isUpper(x)) + key) % 26) + isUpper(x)) : x;
}

const isUpper = (x) => {
  return /[A-Z]/.test(x) ? 65 : 97;
};

File: Exercism Codes/Python/rotational_cipher.py
def rotate(text, key):
    return "".join(rot_letter(x, key) for x in text)

def rot_letter(x: str, key):
    return chr((((ord(x) - is_upper(x)) + key) % 26) + is_upper(x)) if x.isalpha() else x

def is_upper(x):
    return 65 if x.isupper() else 97